sklearn/decomposition/_base.py

```diff
@@ -47,7 +47,9 @@ def get_covariance(self):
             components_ = components_ * xp.sqrt(exp_var[:, np.newaxis])
         exp_var_diff = exp_var - self.noise_variance_
         exp_var_diff = xp.where(
-            exp_var > self.noise_variance_, exp_var_diff, xp.asarray(0.0)
+            exp_var > self.noise_variance_,
+            exp_var_diff,
+            xp.asarray(0.0, device=getattr(exp_var, "device", None)),
```

Comment:
Should we use `device` from `sklearn/utils/_array_api.py` for this (and below)?

Reply:
Sounds good! I am changing it

Yeah, this is somehow unrelated with the introduction of `mps`. It's curious that `cuda` did not complain

Reply:
This is related to `mps` because PyTorch with the MPS device would crash in this case (other PyTorch devices are more lenient for some reasons).

Reply:
To be more precise the crash would happen afterwards, when doing operations that combine tensors from 2 different devices.